pyro/distributions/transforms/affine_autoregressive.py

```diff
@@ -220,6 +223,110 @@ def _inverse_stable(self, y):
+    """
+    An implementation of the bijective transform of Inverse Autoregressive Flow
+    (IAF) that conditions on an additional context variable and uses, by default,
+    Eq (10) from Kingma Et Al., 2016,
+
+        :math:`\\mathbf{y} = \\mu_t + \\sigma_t\\odot\\mathbf{x}`
+
+    where :math:`\\mathbf{x}` are the inputs, :math:`\\mathbf{y}` are the outputs,
+    :math:`\\mu_t,\\sigma_t` are calculated from an autoregressive network on
+    :math:`\\mathbf{x}` and context :math:`\\mathbf{z}\\in\\mathbb{R}^M`, and
+    :math:`\\sigma_t>0`.
+
+    If the stable keyword argument is set to True then the transformation used is,
+
+        :math:`\\mathbf{y} = \\sigma_t\\odot\\mathbf{x} + (1-\\sigma_t)\\odot\\mu_t`
+
+    where :math:`\\sigma_t` is restricted to :math:`(0,1)`. This variant of IAF is
+    claimed by the authors to be more numerically stable than one using Eq (10),
+    although in practice it leads to a restriction on the distributions that can be
+    represented, presumably since the input is restricted to rescaling by a number
+    on :math:`(0,1)`.
+
+    Together with :class:`~pyro.distributions.ConditionalTransformedDistribution`
+    this provides a way to create richer variational approximations.
+
+    Example usage:
+
+    >>> from pyro.nn import ConditionalAutoRegressiveNN
+    >>> input_dim = 10
+    >>> context_dim = 4
+    >>> batch_size = 3
+    >>> hidden_dims = [10*input_dim, 10*input_dim]
+    >>> base_dist = dist.Normal(torch.zeros(input_dim), torch.ones(input_dim))
+    >>> hypernet = ConditionalAutoRegressiveNN(input_dim, context_dim, hidden_dims)
+    >>> transform = ConditionalAffineAutoregressive(hypernet)
+    >>> pyro.module("my_transform", transform)  # doctest: +SKIP
+    >>> z = torch.rand(batch_size, context_dim)
+    >>> flow_dist = dist.ConditionalTransformedDistribution(base_dist,
+    ... [transform]).condition(z)
+    >>> flow_dist.sample(sample_shape=torch.Size([batch_size]))  # doctest: +SKIP
+
+    The inverse of the Bijector is required when, e.g., scoring the log density of a
+    sample with :class:`~pyro.distributions.TransformedDistribution`. This
+    implementation caches the inverse of the Bijector when its forward operation is
+    called, e.g., when sampling from
+    :class:`~pyro.distributions.TransformedDistribution`. However, if the cached
+    value isn't available, either because it was overwritten during sampling a new
+    value or an arbitary value is being scored, it will calculate it manually. Note
+    that this is an operation that scales as O(D) where D is the input dimension,
+    and so should be avoided for large dimensional uses. So in general, it is cheap
+    to sample from IAF and score a value that was sampled by IAF, but expensive to
+    score an arbitrary value.
+
+    :param autoregressive_nn: an autoregressive neural network whose forward call
+        returns a real-valued mean and logit-scale as a tuple
+    :type autoregressive_nn: nn.Module
```

Comment:
I think you need to replace this docstring with a description of the `hypernet` arg?

Reply:
done